File: hu/codes/python/chapter_tree/binary_tree.py
"""
File: binary_tree.py
Created Time: 2022-12-20
Author: a16su (lpluls001@gmail.com)
"""

import sys
from pathlib import Path

sys.path.append(str(Path(__file__).parent.parent))
from modules import TreeNode, print_tree


"""Fő kód"""
if __name__ == "__main__":
    # Bináris fa inicializálása
    # Csomópontok inicializálása
    n1 = TreeNode(val=1)
    n2 = TreeNode(val=2)
    n3 = TreeNode(val=3)
    n4 = TreeNode(val=4)
    n5 = TreeNode(val=5)
    # Csomópontok közötti referenciák (mutatók) felépítése
    n1.left = n2
    n1.right = n3
    n2.left = n4
    n2.right = n5
    print("\nBináris fa inicializálása\n")
    print_tree(n1)

    # Csomópontok beszúrása és törlése
    P = TreeNode(0)
    # P csomópont beszúrása az n1 -> n2 közé
    n1.left = P
    P.left = n2
    print("\nP csomópont beszúrása után\n")
    print_tree(n1)
    # Csomópont törlése
    n1.left = n2
    print("\nP csomópont törlése után\n")
    print_tree(n1)

File: hu/codes/java/chapter_tree/binary_tree.java
/**
 * File: binary_tree.java
 * Created Time: 2022-11-25
 * Author: krahets (krahets@163.com)
 */

package chapter_tree;

import utils.*;

public class binary_tree {
    public static void main(String[] args) {
        /* Bináris fa inicializálása */
        // Csomópontok inicializálása
        TreeNode n1 = new TreeNode(1);
        TreeNode n2 = new TreeNode(2);
        TreeNode n3 = new TreeNode(3);
        TreeNode n4 = new TreeNode(4);
        TreeNode n5 = new TreeNode(5);
        // Hivatkozások (mutatók) felépítése a csomópontok között
        n1.left = n2;
        n1.right = n3;
        n2.left = n4;
        n2.right = n5;
        System.out.println("\nInitialize binary tree\n");
        PrintUtil.printTree(n1);

        /* P csomópont beszúrása n1 -> n2 közé */
        TreeNode P = new TreeNode(0);
        // Csomópont törlése
        n1.left = P;
        P.left = n2;
        System.out.println("\nAfter inserting node P\n");
        PrintUtil.printTree(n1);
        // P csomópont eltávolítása
        n1.left = n2;
        System.out.println("\nAfter removing node P\n");
        PrintUtil.printTree(n1);
    }
}
